### ISSUE TITLE
[Bug]: UI Mode snapshot sometimes displays double

### ISSUE THREAD

**Issue by agg23** [microsoft/playwright#33940]
### Version

1.49.1

### Steps to reproduce

1. Create a test containing
```ts
await page.goto("https://github.com/");
await page.goto("https://github.com/");
```
2. Run the test in UI Mode
3. Hover over the entries for the gotos. Sometimes the second entry shows doubled UI elements, offset slightly

![Image](https://github.com/user-attachments/assets/c6b388a0-11b1-426f-833e-748c8bbf989d)

### Expected behavior

The snapshot display should be clean.

### Actual behavior

![Image](https://github.com/user-attachments/assets/fdadada0-c048-4609-a97d-4a68054148e1)

### Environment

```shell
System:
    OS: Windows 11 10.0.26100
    CPU: (6) x64 Intel(R) Core(TM) i5-9500 CPU @ 3.00GHz
    Memory: 2.76 GB / 15.81 GB
  Binaries:
    Node: 20.17.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.22 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 10.8.2 - C:\Program Files\nodejs\npm.CMD
  IDEs:
    VSCode: 1.95.2 - C:\Users\agastineau\AppData\Local\Programs\Microsoft VS Code\bin\code.CMD
```

**Comment by yury-s** [comment]
Does it reproduce in trace viewer? If so, can you attach trace file?

**Comment by saeed6944** [comment]
> ### Version
> 
> 1.49.1
> 
> ### Steps to reproduce
> 
> 1. Create a test containing
> ```ts
> await page.goto("https://github.com/");
> await page.goto("https://github.com/");
> ```
> 2. Run the test in UI Mode
> 3. Hover over the entries for the gotos. Sometimes the second entry shows doubled UI elements, offset slightly
> 
> ![Image](https://github.com/user-attachments/assets/c6b388a0-11b1-426f-833e-748c8bbf989d)
> 
> 
> ### Expected behavior
> 
> The snapshot display should be clean.
> 
> ### Actual behavior
> 
> ![Image](https://github.com/user-attachments/assets/fdadada0-c048-4609-a97d-4a68054148e1)
> 
> 
> ### Additional context
> 
> _No response_
> 
> ### Environment
> 
> ```shell
> System:
>     OS: Windows 11 10.0.26100
>     CPU: (6) x64 Intel(R) Core(TM) i5-9500 CPU @ 3.00GHz
>     Memory: 2.76 GB / 15.81 GB
>   Binaries:
>     Node: 20.17.0 - C:\Program Files\nodejs\node.EXE
>     Yarn: 1.22.22 - ~\AppData\Roaming\npm\yarn.CMD
>     npm: 10.8.2 - C:\Program Files\nodejs\npm.CMD
>   IDEs:
>     VSCode: 1.95.2 - C:\Users\agastineau\AppData\Local\Programs\Microsoft VS Code\bin\code.CMD
> ```

**Comment by agg23** [comment]
> Does it reproduce in trace viewer? If so, can you attach trace file?

Good point. [trace.zip](https://github.com/user-attachments/files/18096685/trace.zip). It does reproduce in the hosted version of Trace Viewer.
